### ISSUE TITLE
No "Time Remaining" Section on Mobile

### ISSUE THREAD

**Issue by mswider** [chromezoneeths/ethsbell-rewrite#72]
**Description:**
On the desktop version of the site and in the extension, the homepage shows the type of day along with the ongoing class periods and the time left in them. The part that shows the ongoing class periods seems to be either missing or moved to an inaccessible area on mobile. **This effectively makes the main page pointless as it no longer displays any useful information.**

**To Reproduce**
(Steps to reproduce the behavior:
1. Go to [the homepage](https://ethsbell.app)
2. See error (the info isn't there)

**Expected behavior**
The section displaying the ongoing class periods along with their duration is visible. 

**Screenshots:**
Problem: 
![IMG_0556](https://user-images.githubusercontent.com/37093293/131030897-9c3926f1-138e-4b48-96fa-387668de3171.PNG)

Expected: (Making the size the same size as an iPhone 6 in devTools)
_The fact that it works with the same size as mobile seems to indicate that it's not a layout issue_
![Screenshot 2021-08-26 3 03 41 PM](https://user-images.githubusercontent.com/37093293/131030854-bec1ef04-8701-44fd-86c7-0d1357cd4676.png)

**Server Version**
```json
["1.6.7","9dca1ffe517701b68d99344bc2eed3eba1deb3c7","chromezoneeths/ethsbell-rewrite"]
```

**OS/Browser Info**
I'm using an iPhone 6
Chrome: [whatsmybrowser.org/b/LG7T3](https://whatsmybrowser.org/b/LG7T3)
Safari: [whatsmybrowser.org/b/CX7X2](https://whatsmybrowser.org/b/CX7X2)

**Additional context**
I'm pretty sure that this was still working last week because I used to check it on my phone a lot last week and at least a part of this week with no issue.

**Comment by mswider** [comment]
**Update:** My phone does not support [String.prototype.replaceAll()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll) oddly enough. This would mean that any calls to _replaceAll_ on a string would fail, since it would be undefined instead of a function.
**Source:**  In the [Safari 13.1 Release Notes](https://developer.apple.com/documentation/safari-release-notes/safari-13_1-release_notes), support was added for the String.prototype.replaceAll method, meaning that the method still does not exist on Safari versions earlier than 13.1, like my phone, which runs version 12.1.2. The problem still exists in Chrome on my phone because it seems to use parts of Safari, interestingly.

**Problem discovered:**
[A somewhat recent commit to prevent XSS](https://github.com/chromezoneeths/ethsbell-rewrite/commit/e5ccb54f8cdbb2e0bdaedddabcf987f92dc8a7da) used _replaceAll_ to sanitize the period names. However, since my phone doesn't support that method, any use of the function [period_html](https://github.com/chromezoneeths/ethsbell-rewrite/blob/724b8fd9038c1a7e9959a2d8dbf488688aa3cc0e/frontend/global/helpers.js#L55) will likely fail, triggering failures up to the function that resulted in it eventually being called. For example, if the sanitized element returned by the problem function was needed by a certain view, a failure in getting that sanitized element would result in that view not being shown because the function had an error.

I've discovered since filing this issue that the schedule and timeline view are also broken on my phone, displaying no data. The problem I've discovered is probably the cause of both of these issues, since **both views result in the problem function being called.**
Unfortunately, even if the issue I've found were to be resolved, I'm not sure that the bug I filed (data is missing at the homepage) would be fixed, since I couldn't find the problem function referenced in the code for it, meaning that the error is still probably caused by something else.
